### PR TITLE
feat: implement a few more functions as part of supporting query from parquet file

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes #

Second PR for #1082

I think left over:
- [ ] implement read_filter for parquet
- [ ] Only query parquet files when corresponding RB chunks do not exists

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: data_types/src/timestamp.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ pub struct TimestampRange {
 
 impl TimestampRange {
     pub fn new(start: i64, end: i64) -> Self {
-        debug_assert!(end > start);
+        debug_assert!(end >= start);
```
Comment (alamb) [inline data_types/src/timestamp.rs:15]: 👍

File: server/src/db/chunk.rs [hunk 2]
```diff
@@ -343,8 +343,12 @@ impl PartitionChunk for DBChunk {
                         })?,
                 ))
             }
-            Self::ParquetFile { .. } => {
-                unimplemented!("parquet file not implemented for column_names")
+            Self::ParquetFile { chunk, .. } => {
+                if !predicate.is_empty() {
+                    // TODO: Support predicates when MB supports it
```
Comment (alamb) [inline server/src/db/chunk.rs:348]: This comment looks out of date (it is about Parquet rather than the MB I think)

**Review by alamb — APPROVED** [review]
Looks good to me -- I can't wait to see this all hooked up in `query_tests` !

File: server/src/db/chunk.rs (hunk 2 above)
Comment (NGA-TRAN) [reply server/src/db/chunk.rs:348]: I added it because MB does not do it so I kind of wait until we support this in MB :)

**Review by e-dard — APPROVED** [review]
LGTM 👍

**Pull request merged**